scripts/o3de/o3de/export_project.py

```diff
@@ -136,3 +283,402 @@ def add_args(subparsers) -> None:
+    @param required:            If true, check if the asset bundler actually exists on file at the expected location, and raise an error if not
+    @return: Path to the asset processor tool
+    """
+    asset_processor_batch_path = tools_build_path / f'bin/profile/AssetProcessorBatch{EXECUTABLE_EXTENSION}'
+    if required and not asset_processor_batch_path.is_file():
+        raise ExportProjectError(f"Missing the 'AssetProcessorBatch' tool, expected at '{asset_processor_batch_path}'")
+    return asset_processor_batch_path
+
+
+def get_asset_bundler_batch_path(tools_build_path: pathlib.Path,
+                                 required: bool = False) -> pathlib.Path:
+    """
+    Get the expected path to the asset bundler tool
+
+    @param tools_build_path:    The tools (cmake) build path to locate AssetBundlerBatch
+    @param required:            If true, check if the asset bundler actually exists on file at the expected location, and raise an error if not
+    @return: Path to the asset bundler tool
+    """
+
+    asset_bundler_batch_path = tools_build_path / f'bin/profile/AssetBundlerBatch{EXECUTABLE_EXTENSION}'
+    if required and not asset_bundler_batch_path.is_file():
+        raise ExportProjectError(f"Missing the 'AssetBundlerBatch' tool, expected at '{asset_bundler_batch_path}'")
+    return asset_bundler_batch_path
+
+
+def build_assets(ctx: O3DEScriptExportContext,
+                 tools_build_path: pathlib.Path,
+                 fail_on_ap_errors: bool,
+                 logger: logging.Logger = None) -> None:
+    """
+    Build the assets for the project
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to locate AssetProcessorBatch
+    @param fail_on_ap_errors:   Option to fail the whole process if an error occurs during asset processing
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Make sure `AssetProcessorBatch` is available
+    asset_processor_batch_path = get_asset_processor_batch_path(tools_build_path, required=True)
+    if not asset_processor_batch_path.exists():
+        raise ExportProjectError("Missing AssetProcessorBatch. The pre-requisite tools must be built first.")
+
+    # Build the project assets with the {project_name}.Assets custom target
+    if logger:
+        logger.info(f"Processing assets for {ctx.project_name}")
+
+    cmake_build_assets_command = [asset_processor_batch_path, "--project-path", ctx.project_path]
+    ret = process_command(cmake_build_assets_command,
+                          cwd=ctx.engine_path if ctx.is_engine_centric else ctx.project_path)
+    if ret != 0:
+        if fail_on_ap_errors:
+            raise ExportProjectError(f"Error building assets for project {ctx.project_name}.")
+        else:
+            if logger:
+                logger.warning("Some assets failed to processed.")
+    if logger:
+        logger.info(f"Completed processing assets for {ctx.project_name}!")
```

Comment:
This seems like it should return the AssetProcessorBatch return code at the end of this function when `fail_on_ap_errors` is false.